### PR TITLE
Fix installing required file bundles

### LINKED ISSUE #6046: Plugins with related file bundles don't install with latest Hub
Terminal 1:

```sh
$ git clone git@github.com:meltano/hub.git
$ cd hub
$ git checkout develop
# install all dependencies
$ bundle exec jekyll serve --livereload 
```

Terminal 2:

```sh
$ meltano init airflow-test
$ cd airflow-test
$ meltano config meltano set hub_url http://localhost:4000
$ meltano --log-level=debug add orchestrator airflow
2022-06-07T03:01:13.107560Z [info     ] Environment 'dev' is active
2022-06-07T03:01:13.119717Z [debug    ] Creating engine <meltano.core.project.Project object at 0x111d5dc10>@sqlite:////Users/douwemaan/Development/meltano-projects/airflow-test/.meltano/meltano.db
2022-06-07T03:01:13.208891Z [debug    ] Lockfile is feature-flagged    status=False
2022-06-07T03:01:13.214385Z [debug    ] Starting new HTTP connection (1): localhost:4000
2022-06-07T03:01:13.216051Z [debug    ] http://localhost:4000 "GET /meltano/api/v1/plugins/orchestrators/index HTTP/1.1" 200 221
2022-06-07T03:01:13.218166Z [debug    ] http://localhost:4000 "GET /meltano/api/v1/plugins/orchestrators/airflow--apache HTTP/1.1" 200 1287
2022-06-07T03:01:13.240066Z [debug    ] Found plugin parent            parent=airflow plugin=airflow source=hub
Added orchestrator 'airflow' to your Meltano project
Variant:        apache (default)
Repository:     https://github.com/apache/airflow
Documentation:  https://docs.meltano.com/guide/orchestration

2022-06-07T03:01:13.317088Z [debug    ] Starting new HTTPS connection (1): discovery.meltano.com:443
2022-06-07T03:01:13.419546Z [debug    ] https://discovery.meltano.com:443 "GET /discovery.yml HTTP/1.1" 200 22755
2022-06-07T03:01:13.813422Z [debug    ] Lockfile is feature-flagged    status=False
2022-06-07T03:01:13.815892Z [debug    ] http://localhost:4000 "GET /meltano/api/v1/plugins/files/index HTTP/1.1" 200 2299
2022-06-07T03:01:13.816194Z [debug    ] File bundle 'airflow' is not known to Meltano 
Traceback (most recent call last):
  File "/Users/douwemaan/Development/meltano/src/meltano/core/hub/client.py", line 177, in find_definition
    plugin = plugins[plugin_name]
KeyError: 'airflow'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/douwemaan/Development/meltano/src/meltano/cli/__init__.py", line 51, in main
    cli(obj={"project": None})
  File "/Users/douwemaan/.pyenv/versions/3.9.11/envs/meltano-dev/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/douwemaan/.pyenv/versions/3.9.11/envs/meltano-dev/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/douwemaan/.pyenv/versions/3.9.11/envs/meltano-dev/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/douwemaan/.pyenv/versions/3.9.11/envs/meltano-dev/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/douwemaan/.pyenv/versions/3.9.11/envs/meltano-dev/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/douwemaan/Development/meltano/src/meltano/cli/params.py", line 23, in decorate
    return func(*args, **kwargs)
  File "/Users/douwemaan/Development/meltano/src/meltano/cli/params.py", line 56, in decorate
    func(project, *args, **kwargs)
  File "/Users/douwemaan/.pyenv/versions/3.9.11/envs/meltano-dev/lib/python3.9/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/douwemaan/Development/meltano/src/meltano/cli/add.py", line 129, in add
    required_plugins = add_required_plugins(
  File "/Users/douwemaan/Development/meltano/src/meltano/cli/utils.py", line 414, in add_required_plugins
    required_plugins = add_service.add_related(
  File "/Users/douwemaan/Development/meltano/src/meltano/core/project_add_service.py", line 118, in add_related
    plugin = self.add(plugin_ref.type, plugin_ref.name)
  File "/Users/douwemaan/Development/meltano/src/meltano/core/project_add_service.py", line 68, in add
    self.plugins_service.ensure_parent(plugin)
  File "/Users/douwemaan/Development/meltano/src/meltano/core/project_plugins_service.py", line 537, in ensure_parent
    plugin.parent = self.get_parent(plugin)
  File "/Users/douwemaan/Development/meltano/src/meltano/core/project_plugins_service.py", line 520, in get_parent
    parent, source = self.find_parent(plugin)
  File "/Users/douwemaan/Development/meltano/src/meltano/core/project_plugins_service.py", line 509, in find_parent
    raise error
  File "/Users/douwemaan/Development/meltano/src/meltano/core/project_plugins_service.py", line 504, in find_parent
    return (self._get_parent_from_hub(plugin), DefinitionSource.HUB)
  File "/Users/douwemaan/Development/meltano/src/meltano/core/project_plugins_service.py", line 441, in _get_parent_from_hub
    return self.hub_service.get_base_plugin(plugin, variant_name=plugin.variant)
  File "/Users/douwemaan/Development/meltano/src/meltano/core/plugin_discovery_service.py", line 136, in get_base_plugin
    plugin = project_plugin.custom_definition or self.find_definition(
  File "/Users/douwemaan/Development/meltano/src/meltano/core/hub/client.py", line 179, in find_definition
    raise PluginNotFoundError(
meltano.core.plugin.error.PluginNotFoundError: File bundle 'airflow' is not known to Meltano

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/douwemaan/Development/meltano/src/meltano/cli/__init__.py", line 59, in main
    raise CliError(str(err)) from err
meltano.cli.utils.CliError: File bundle 'airflow' is not known to Meltano
File bundle 'airflow' is not known to Meltano
```

Related to https://github.com/meltano/meltano/pull/5963 and https://github.com/meltano/hub/pull/538

The problem seems to be that Meltano tries to install the `airflow` file bundle, even though this has now been renamed to `files-airflow` in the Hub.

Same result with `dbt`.

### PR DESCRIPTION
**Pull request by DouweM**
Fixes https://github.com/meltano/meltano/issues/6046

Even though `required` was set, `meltano add orchestrator airflow` would look for plugins with the same `namespace` in `discovery.yml`, find the `airflow` file bundle there, and try to add it. Then during adding that plugin, inside the `disallow_discovery_yaml` context, Meltano would look for that plugin on the Hub, and not find it there (as it had been renamed in https://github.com/meltano/hub/pull/538) and fail with the error shown in https://github.com/meltano/meltano/issues/6046.

In other words: to find related plugins, Meltano only looks at discovery, and then to add those related plugins, it only looks at the Hub. So if a plugin exists in the one but not the other, things break.

Now, if `required` is set, `namespace`-based related plugins are ignored. For some reason, `add_required` doesn't seem to have been called at all before.

### CODE REVIEW

File: src/meltano/core/project_add_service.py [hunk 1]
```diff
@@ -106,9 +112,12 @@ def add_related(self, *args, **kwargs):
         Returns:
             The added plugins.
         """
+        if project_plugin.requirements:
+            return self.add_required(project_plugin)
```
Comment (DouweM) [inline src/meltano/core/project_add_service.py:116]: `add_required` was never actually being called before, so the `required` property was being ignored, and `meltano add orchestrator airflow` would look for plugins with the same `namespace`, find `airflow` in `discovery.yml` (but not on the Hub because it had been renamed to `files-airflow` there), and try to install it. Then at some point it would apparently look for that plugin from `discovery.yml` on the Hub, and not find it there and fail with the error shown in https://github.com/meltano/meltano/issues/6046.

Now, if `required` is set, `namespace`-based related plugins are ignored.

Comment (DouweM) [inline src/meltano/core/project_add_service.py:116]: @edgarrmondragon Now that all plugins on the hub have `requires` (https://github.com/meltano/hub/pull/538), do you think we can remove the whole discovery/namespace-based `add_related` code path? I added this in with a conditional just to be sure.

**Comment by DouweM** [comment]
@edgarrmondragon points out this may be a dupe of https://github.com/meltano/meltano/pull/5999.

**Comment by DouweM** [comment]
@edgarrmondragon Do you think anything from this  PR is worth saving now that #5999 is merged?

**Comment by edgarrmondragon** [comment]
@DouweM I went ahead and removed lingering `related` code paths and would like to save some of the updates from this PR. I'll merge once tests pass.

**Review by aaronsteers — APPROVED** [review]
@edgarrmondragon - Per your last comment, I know you are probably still working on this.

Pre-approving so you can merge when tests pass.
